### DISCUSSION TITLE
Can't create multiple tables using automap without inheritance

### DISCUSSION THREAD

**Discussion by JohnnyRacer** [sqlalchemy/sqlalchemy#9360] [Usage Questions] [answered]
### Describe the bug

Hello , I would like to use `automap_base` from the `automap` module to create multiple tables that have no relationships between them. However I am not able find a way to do this as `SQLAlchemy` always complains that I need to specify a inheritance condition or foreign key relationship. I tried to set `inherit_condition` to `None` in the ` __mapper_args__ ` but this does nothing. There is very little information in the docs about `inherit_condition` option as well. The error occurs even I use a seperate `automap_base` object for each of the table schemas. 

### SQLAlchemy Version in Use

 2.0.4 

### DBAPI (i.e. the database driver)

asyncpg

### Database Vendor and Major Version

PostgreSQL 12,

### Python Version

3,8.13

### Operating system

Linux

### To Reproduce

```python
from sqlalchemy.ext.automap import automap_base
from sqlalchemy import Column, String, DateTime, create_engine

engine = create_engine('connection string')

Base = automap_base()

class TestData(Base):
    __tablename__ = 'testdata'
    id = Column('id',Integer,autoincrement=True, primary_key=True)
    time_created = Column('createdAt' ,DateTime(timezone=True), server_default=func.now())
    time_updated = Column('updatedAt' ,DateTime(timezone=True), onupdate=func.now())

    dataid = Column('dataid', String, nullable=False)
    data = Column('data', String, nullable=False)

    __mapper_args__ = {
        "inherit_condition" : None,
        "polymorphic_identity": "testdata",
    }

class TestUser(Base):
    __tablename__ = 'testuser'
    id = Column('id',Integer,autoincrement=True, primary_key=True)
    time_created = Column('createdAt' ,DateTime(timezone=True), server_default=func.now())
    time_updated = Column('updatedAt' ,DateTime(timezone=True), onupdate=func.now())

    username = Column('username', String, nullable=False)
    userid = Column('userid', String, nullable=False)

    __mapper_args__ = {
        "inherit_condition" : None,
        "polymorphic_identity": "testuser",
    }

Base.metadata_create_all(engine)
Base.prepare(engine)
```

### Error

```
File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/mapper.py", line 1240, in _configure_inheritance
raise sa_exc.NoForeignKeysError(
sqlalchemy.exc.NoForeignKeysError: Can't determine the inherit condition between inherited table 'testdata' and inheriting table 'testUser'; tables have no foreign key relationships established.  Please ensure the inheriting table has a foreign key relationship to the inherited table, or provide an 'on clause' using the 'inherit_condition' mapper argument.
```

**Comment by zzzeek** [comment, marked as answer]
hi -

this is not a bug, as this exception is standard when two classes are given in a joined inheritance scenario without a means given of the two tables having a join condition.    If the two tables have no foreign key constraints between them, this is the error raised, and [inherit_condition](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.Mapper.params.inherit_condition) needs to specify such a condition, such as `inherit_condition=TestData.id == TestUser.id`.   For a complete example see https://docs.sqlalchemy.org/en/20/orm/extensions/automap.html#relationships-with-inheritance .

the example case as shown can't possibly raise this error because the two classes `TestData` and `TestUser` are not in an inheritance configuration, and automap does not decide that two tables would be in joined inheritance like that.   You'd see the classes more like `TestUser(TestData)`.

**Comment by JohnnyRacer** [comment]
@zzzeek 
I actually attached the wrong code snippet which actually was giving the correct desired behavior without the `NoForeignKeysError`. 
However I do not understand why this snippet below would give this error, since I am not directly inheriting the classes between the two. 
```python
TestData.metadata_create_all(engine)
TestData.prepare(engine)

TestUser.metadata_create_all(engine)
TestUser.prepare(engine)
```
The above class declaration is the same but running this snippet will give the same ` NoForeignKeysError`, is this intentional behavior?

**Reply by zzzeek** [reply]
OK the thing you are showing is well within  "there's no possible way that could happen", yet it happens, so #9367 is created for that.

however for your immediate case, don't call prepare() from a specific class like that, call `Base.prepare()`.  the bug has something to do with calling `prepare()` from a subclass.

**Reply by JohnnyRacer** [reply]
Sorry, I had included the following example to illustrate what I had done before realizing my mistake.  I was kind of misled by the documentation into thinking that for each table schema class must be inherited from `automap_base` to allow for creation and access, but it was actually the other way around. I think it would be helpful if there were more clearer points on creating multiple tables with `automap_base` in the documentation.

**Reply by zzzeek** [reply]
you found a bug!   time to celebrate  :tada:
